tests/FINDOLOGIC/Api/Tests/RequestBuilders/Json/SuggestionRequestBuilderTest.php

```diff
@@ -0,0 +1,192 @@
+        } catch (ParamNotSetException $e) {
+            $this->assertEquals('Required param query is not set.', $e->getMessage());
+        }
+
+        $this->httpClientMock->method('request')->willReturn($this->responseMock);
+        $this->responseMock->method('getBody')->willReturn($this->streamMock);
+        $this->responseMock->method('getStatusCode')->willReturn(200);
+        $this->streamMock->method('getContents')
+            ->willReturnOnConsecutiveCalls(
+                $this->rawMockResponse,
+                $this->rawMockResponse
+            );
+
+        $suggestionRequestBuilder->setQuery('something');
+        $client->send($suggestionRequestBuilder);
+    }
+
+    /**
+     * @dataProvider queryProvider
+     * @param string $expectedQuery
+     */
+    public function testSetQueryWillSetItInAValidFormat($expectedQuery)
+    {
+        $expectedParameter = 'query';
+
+        $searchRequestBuilder = new SuggestRequestBuilder();
+        $this->setRequiredParamsForSuggestionRequestBuilder($searchRequestBuilder);
+
+        $searchRequestBuilder->setQuery($expectedQuery);
+        $params = $searchRequestBuilder->getParams();
+        $this->assertArrayHasKey($expectedParameter, $params);
+        $this->assertEquals($expectedQuery, $params[$expectedParameter]);
+    }
+
+    /**
+     * @dataProvider invalidQueryProvider
+     * @param mixed $invalidQuery
+     */
+    public function testSetQueryWillThrowAnExceptionWhenSubmittingInvalidQueries($invalidQuery)
+    {
+        $this->expectException(InvalidParamException::class);
+        $this->expectExceptionMessage('Parameter query is not valid.');
+
+        $searchRequestBuilder = new SuggestRequestBuilder();
+        $this->setRequiredParamsForSuggestionRequestBuilder($searchRequestBuilder);
+
+        $searchRequestBuilder->setQuery($invalidQuery);
+    }
+
+    /**
+     * @dataProvider autocompleteBlocksProvider
+     * @param string $autocompleteBlock
+     */
+    public function testAddAutocompleteBlocksWillSetItInAValidFormat($autocompleteBlock)
+    {
+        $expectedParameter = 'autocompleteblocks';
+
+        $searchRequestBuilder = new SuggestRequestBuilder();
+        $this->setRequiredParamsForSuggestionRequestBuilder($searchRequestBuilder);
+
+        $searchRequestBuilder->addAutocompleteBlocks($autocompleteBlock);
+        $params = $searchRequestBuilder->getParams();
+        $this->assertArrayHasKey($expectedParameter, $params);
+        $this->assertEquals($autocompleteBlock, $params[$expectedParameter]);
+    }
+
+    /**
+     * @dataProvider invalidAutocompleteBlocksProvider
+     * @param mixed $invalidAutocompleteblock
+     */
+    public function testInvalidShopurlWillThrowAnException($invalidAutocompleteblock)
+    {
+        $this->expectException(InvalidParamException::class);
+        $this->expectExceptionMessage('Parameter autocompleteblocks is not valid.');
```

Comment:
Since this test is about shop URL, is this message correct?

Reply:
Yes everything seems to be fine, just the method name is not. I probably copied this test from the XmlRequestBuilder tests and forgot to rename the method itself.